src/EnergyPlus/FileSystem.hh

```diff
@@ -320,6 +313,43 @@ namespace FileSystem {
         }
     }
 
+    std::string toString(fs::path const &p);
+
+    std::string toGenericString(fs::path const &p);
+
+    fs::path appendSuffixToPath(fs::path const& outputFilePrefixFullPath, const std::string &suffix);
+
 } // namespace FileSystem
 } // namespace EnergyPlus
+
+// Add a custom formatter for fmt
+template <> struct fmt::formatter<fs::path>
+{
+    // Presentation format: 's' - string, 'g' - generic_string.
+    char presentation = 's';
+
+    // Parses format specifications of the form ['s' | 'g'].
+    constexpr auto parse(format_parse_context &ctx) -> decltype(ctx.begin())
+    {
+        // Parse the presentation format and store it in the formatter:
+        auto it = ctx.begin(), end = ctx.end();
+        if (it != end && (*it == 's' || *it == 'g')) {
+            presentation = *it++;
+        }
+
+        // Check if reached the end of the range:
+        if (it != end && *it != '}') {
+            throw format_error("invalid format");
+        };
+
+        // Return an iterator past the end of the parsed range:
+        return it;
+    }
+
+    template <typename FormatContext> auto format(const fs::path &p, FormatContext &ctx) -> decltype(ctx.out())
+    {
+        return format_to(ctx.out(), "{}", presentation == 'g' ? EnergyPlus::FileSystem::toGenericString(p) : EnergyPlus::FileSystem::toString(p));
+    }
+};
```

Comment:
new formatter for fs::path, instead of using the std::string one.

Two presentation types are accepted: `s` (default), and `g`.

`s` uses toString (and is similar to fs::path::string() but with narrowing) and `g` uses toGenericString (and is sililar to fs::path::generic_string())

Reply:
This looks great to me.  I'm like 23% surprised you had to make your own formatter for this.  So 77% not surprised because of wide char nonsense.